index.js

```diff
@@ -30,19 +22,14 @@ function JMON(data) {
   this.initialData = data;
   this.data = _.cloneDeep(this.initialData);
 
-  if (isObject){
+  Object.keys(this.data).forEach((key) => {
```

Comment:
On set function we have to handle creation of key if not present  and then move the responsilbities of creating further  data to its children.

Reply:
> we will handle that in set.

that will be exact copy of this code right?

> On set function we have to handle creation of key if not present and then move the responsilbities of creating further data to its children.

set will not handle creation, for creation we will have `push` if key is not present it will throw error saying use push for creating new

Reply:
i was thinking like we dont create a file for every loop we perform right... I feel like this piece of code is also  loop that have to be done at every where needed. I think it will not break the DRY concept also.

Reply:
> new

I guess only parent can set value of child right. Then if we provide Json also then that would be converted into JMON and assign it to  the key .. while all the data inside should be handled in constructor. We might not need foreach in set i suppose.

Reply:
@arshadkazmi42

Reply:
true!!